File: user-service/src/main/java/com/phorus/userservice/services/UserService.java
package com.phorus.userservice.services;

import com.phorus.userservice.model.dbentities.User;
import com.phorus.userservice.model.dtos.UserDTO;


public interface UserService {

    User findById(Long userId);

    Long create(User user);

    void update(Long userId, UserDTO userDTO);

    void deleteById(Long userId);


    Long getAmountOfPets(Long userId);
}

File: user-service/src/main/kotlin/com/phorus/userservice/services/UserService.kt
package com.phorus.userservice.services

import com.phorus.userservice.model.dbentities.User
import com.phorus.userservice.model.dtos.UserDTO
import reactor.core.publisher.Mono

interface UserService {

    suspend fun findById(userId: Long): User

    suspend fun create(user: User): Long

    suspend fun update(userId: Long, userDTO: UserDTO)

    suspend fun deleteById(userId: Long)

    suspend fun getAmountOfPets(userId: Long): Long
}